python/cudf/cudf/core/dataframe.py

```diff
@@ -4385,32 +4385,41 @@ def from_gpu_matrix(
 
         if data.ndim != 2:
             raise ValueError(
-                "matrix dimension expected 2 but found {!r}".format(data.ndim)
+                f"matrix dimension expected 2 but found {data.ndim}"
             )
 
         if columns is None:
             names = [i for i in range(data.shape[1])]
         else:
             if len(columns) != data.shape[1]:
-                msg = "columns length expected {!r} but found {!r}"
-                raise ValueError(msg.format(data.shape[1], len(columns)))
+                raise ValueError(
+                    f"columns length expected {data.shape[1]} but \
+                        found {len(columns)}"
+                )
             names = columns
 
-        if index is not None and len(index) != data.shape[0]:
-            msg = "index length expected {!r} but found {!r}"
-            raise ValueError(msg.format(data.shape[0], len(index)))
+        if (
+            index is not None
+            and not isinstance(index, (str, int))
+            and len(index) != data.shape[0]
+        ):
+            raise ValueError(
+                f"index length expected {data.shape[0]} but found {len(index)}"
+            )
 
         df = DataFrame()
         data = cupy.asfortranarray(cupy.asarray(data))
         for i, k in enumerate(names):
-            df[k] = Series(data[:, i], nan_as_null=nan_as_null)
+            df._data[k] = as_column(data[:, i], nan_as_null=nan_as_null)
 
         if index is not None:
-            if isinstance(index, cudf.Index):
-                indices = index
+            if isinstance(index, (str, int)):
+                index = as_index(df[index], name=None)
```

Comment:
It doesn't retain the name here?

Reply:
Since we are actually trying to convert a Series to an index, it is carrying over Series `name` to Index `name` and that gets assigned to the DataFrame which has an index with `name`. Since we are targetting this to be used as an internal API for `DataFrame.__init__` in a future PR, I had it removed.

But to let or not let the Series name propagate to Index of the DataFrame is something we may want to decide? Because `set_index` actually seems to carry the series name to the index: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.set_index.html

Reply:
Lets propagate the name, it feels really weird that you're passing `index="index_col"` and then the index comes back with name `None`.

Reply:
Sure, made the change 👍